radicle-httpd/src/api/json.rs

```diff
@@ -136,33 +142,24 @@ fn name_in_path(path: &str) -> &str {
 
 #[derive(Serialize)]
 #[serde(rename_all = "camelCase")]
-struct Comment {
+struct Comment<'a> {
     id: CommentId,
     author: Author,
-    body: String,
-    reactions: [String; 0],
+    body: &'a str,
+    reactions: Vec<(&'a ActorId, &'a Reaction)>,
```

Comment:
I'm curious as to why the lifetimes are needed :)

Reply:
The lifetimes are needed due to `thread.reactions(&EntryId)` returning `Iterator<Item = (&ActorId, &Reaction)>` so AFAIK we either clone the tuples by mapping over them, or we pass these lifetimes into the `Comment` struct.
But maybe I'm missing something, I'm a lifetime amateur hehe

Reply:
And the `body: &'a str` lifetime I added to avoid converting `comment.body()` into a `String`

Reply:
The lifetimes are needed due to thread.reactions(&EntryId) returning Iterator<Item = (&ActorId, &Reaction)> so AFAIK we either clone the tuples by mapping over them, or we pass these lifetimes into the Comment struct.
But maybe I'm missing something, I'm a lifetime amateur hehe
And the body: &'a str lifetime I added to avoid converting comment.body() into a String

Reply:
Cool, ya that makes sense :+1: